### DISCUSSION TITLE
dock api support

### DISCUSSION THREAD

**Discussion by bobzer** [JackJPowell/hass-unfoldedcircle#39] [Ideas]
Hi,

It would be great if the integration support also the dock to directly blast ir without the remote.
Would it be possible or is it better to have a different integration just for the dock ?
use case exemple : 
I just broked my remote and I would love to still be able to use the ir blaster from the dock through HA.

Anyway, thanks for you great work.

Best regards

**Comment by JackJPowell** [comment]
I saw the picture on discord. I'm sorry man. It feels like the remote reaper comes for us all eventually. 

So the bad news is that the dock itself doesn't expose it's on api per se. All the brains are in the remote. But if your remote is still functional as it powers on and connects to wifi, you could still issue all the same commands including sending IR commands through the dock. In other words, the dock is fully supported in HA, but still requires the remote to access it.

**Comment by bobzer** [comment]
thank you for your consideration,
thanks fully it seem to be only the screen, problem is that since the pin constantly change I can't go in the web configurator. So I don't know how to send command (I don't remember the name of device nor the name of command). I do have the pronto code.
Would it be possible to send a ir command with the pronto code as a data payload ?

**Comment by bobzer** [comment]
hi guys, 
back in the game the remote work perfectly but I'm stil strugling to send ir command from home assistant.
so different variation of : 
`service: remote.send_command
data:
  device: tele
  command: POWER_TOGGLE
target:
  entity_id: remote.remote_two_remote`

I took the info from : 
![image](https://github.com/user-attachments/assets/39807694-0f89-4e87-8b4e-561442317a92)

I tried everything I thought, like name, id, name of ir dataset 
I'm out of option (BTW no error in the journal and it works from the remote itself)
could you help me please ?

**Reply by JackJPowell** [reply]
Hey, glad you got it fixed! I'll take a look at least by Friday and see what's going on.

**Reply by JackJPowell** [reply]
This one is probably obvious but double check your remote entity id is what you see in home assistant. Yours matches my example so just making sure. 

Your command I think is correct. The device is where I'd double check first. Take a look at the tip on my readme. I'll dig deeper tomorrow

**Reply by bobzer** [reply]
Thank you for your time, I'm not sure what entity_id to double check ? 
![image](https://github.com/user-attachments/assets/0422e759-84dd-4cb1-a293-2d6631b36682)
is it the name of the id ?

**Reply by JackJPowell** [reply]
Your entity is correct and it looks like everything else you have is correct too. 

Here it is in the developer tools and right below is the yaml for the same thing:
![Screenshot 2024-11-07 at 7 27 37 AM](https://github.com/user-attachments/assets/e2a087d4-0593-49e3-b4d5-4f514b510156)

```
service: remote.send_command
data:
  device: Receiver
  command: Power
target:
  entity_id: remote.remote_two_remote
```

And here is where I got the Device and Command just like you. 
![Screenshot 2024-11-07 at 7 30 00 AM](https://github.com/user-attachments/assets/03c1cda3-9f8f-48cc-ae82-5844fee78930)

In other words, it looks like you are doing everything right. I still have some other things to look at but I couldn't do them really quick so it will be later today.

**Reply by bobzer** [reply]
Thank you to take the time, I'm glad that it was not a stupid mistake on my side because, but hope it's not a big problem and don't hesitate if there is anything I could test or help diagnose.

**Reply by JackJPowell** [reply]
Finally got to test it out on my end but it worked fine. The ir signal is very weak but since you said it worked from the remote, that can't be it. Let me see if I put any debug statements to help out.

**Reply by bobzer** [reply]
Can i do something to test/debug, help ?
thanks

**Reply by JackJPowell** [reply]
Yes! I sent you a message on discord. Let's take a look
